gpiozero/output_devices.py

```diff
@@ -898,3 +898,313 @@ def stop(self):
         self.forward_device.off()
         self.backward_device.off()
 
+
+class Servo(SourceMixin, CompositeDevice):
+    """
+    Extends :class:`CompositeDevice` and represents a PWM-controlled servo
+    motor connected to a GPIO pin.
+
+    Connect a power source (e.g. a battery pack or the 5V pin) to the power
+    cable of the servo (this is typically colored red); connect the ground
+    cable of the servo (typically colored black or brown) to the negative of
+    your battery pack, or a GND pin; connect the final cable (typically colored
+    white or orange) to the GPIO pin you wish to use for controlling the servo.
+
+    The following code will make the servo move between its minimum, maximum,
+    and mid-point positions with a pause between each::
+
+        from gpiozero import Servo
+        from time import sleep
+
+        servo = Servo(17)
+        while True:
+            servo.min()
+            sleep(1)
+            servo.mid()
+            sleep(1)
+            servo.max()
+            sleep(1)
+
+    :param int pin:
+        The GPIO pin which the device is attached to. See :ref:`pin_numbering`
+        for valid pin numbers.
+
+    :param float initial_value:
+        If ``0`` (the default), the device's mid-point will be set
+        initially.  Other values between -1 and +1 can be specified as an
+        initial position. ``None`` means to start the servo un-controlled (see
+        :attr:`value`).
+
+    :param float min_pulse_width:
+        The pulse width corresponding to the servo's minimum position. This
+        defaults to 1ms.
+
+    :param float max_pulse_width:
+        The pulse width corresponding to the servo's maximum position. This
+        defaults to 2ms.
+
+    :param float frame_width:
+        The length of time between servo control pulses measured in seconds.
+        This defaults to 20ms which is a common value for servos.
+    """
+    def __init__(
+            self, pin=None, initial_value=0.0,
+            min_pulse_width=1/1000, max_pulse_width=2/1000,
+            frame_width=20/1000):
+        if min_pulse_width >= max_pulse_width:
+            raise ValueError('min_pulse_width must be less than max_pulse_width')
+        if max_pulse_width >= frame_width:
+            raise ValueError('max_pulse_width must be less than frame_width')
+        self._frame_width = frame_width
+        self._min_dc = min_pulse_width / frame_width
+        self._dc_range = (max_pulse_width - min_pulse_width) / frame_width
+        self._min_value = -1
+        self._value_range = 2
```

Comment:
Looks like these last two variables never change? Do they need to be instance variables? Perhaps instead you could add some static functions which convert a -1 -> +1 value into a 0 -> +1 value, and vice-versa?